content/en/docs/ambient/extensibility/wasm-module-distribution/index.md

```diff
@@ -0,0 +1,162 @@
+---
+title: Distributing WebAssembly Modules
+description: Describes how to make remote WebAssembly modules available for ambient mode.
+weight: 1
+keywords: [extensibility,Wasm,WebAssembly,Ambient]
+owner: istio/wg-policies-and-telemetry-maintainers
+test: no
+status: Alpha
+---
+
+{{< boilerplate alpha >}}
+
+Istio provides the ability to [extend its functionality using WebAssembly (Wasm)](/blog/2020/wasm-announce/).
+One of the key advantages of Wasm extensibility is that extensions can be loaded dynamically at runtime. This document outlines the testing process for the implementation of Wasm features in Ambient mode within Istio. In Ambient mode, Wasm configuration must be applied to the waypoint proxy deployed in each namespace, instead of to individual sidecars. This approach is essential due to the absence of sidecars in Ambient mode, which is a key distinction from previous configurations.
+
+## Install Ambient Mesh and deploy test applications
+
+Follow the [Ambient Getting Started Guide](docs/ambient/getting-started/#download) to install Istio in ambient mode. Deploy the [sample applications](docs/ambient/getting-started/#bookinfo) required for testing the Wasm behavior. Make sure to add the test [applications to ambient mesh](docs/ambient/getting-started/#addtoambient) before proceeding further.
+
+## Apply Wasm configuration at the Gateway
+
+### Configure WasmPlugin for Gateway
+
+In this example, you will add a HTTP [Basic auth module](https://github.com/istio-ecosystem/wasm-extensions/tree/master/extensions/basic_auth) to your mesh. You will configure Istio to pull the Basic auth module from a remote image registry and load it. It will be configured to run on calls to `/productpage`. Steps are more or less similar as [Istio / Distributing WebAssembly Modules](docs/tasks/extensibility/wasm-module-distribution/), only difference being the usage of `targetRef` instead of `labelSelectors` in WasmPlugin.
+
+To configure a WebAssembly filter with a remote Wasm module, create a `WasmPlugin` resource targeting the `bookinfo-gateway`:
+
+{{< text bash >}}
+$ kubectl get gateway
+NAME               CLASS            ADDRESS        PROGRAMMED   AGE
+bookinfo-gateway   istio            172.18.7.110   True         23h
+{{< /text >}}
+
+{{< text bash >}}
+$ kubectl apply -f - <<EOF
+apiVersion: extensions.istio.io/v1alpha1
+kind: WasmPlugin
+metadata:
+  name: basic-auth-at-gateway
+spec:
+  targetRef:
```

Comment:
maybe no need to show wasplugin at gateway, as that is not new and should NOT be unique to ambient.

Reply:
> maybe no need to show wasplugin at gateway, as that is not new and should NOT be unique to ambient.

It is retained. Should we remove it and add it to the original Wasm document instead of having it here ?